server/pipeline/stepStatus.go

```diff
@@ -34,14 +34,10 @@ func UpdateStepStatus(store model.UpdateStepStore, step *model.Step, state rpc.S
 		if state.ExitCode == 137 {
 			step.State = model.StatusKilled
 		}
-	} else {
+	} else if step.Stopped == 0 {
 		step.Started = state.Started
 		step.State = model.StatusRunning
 	}
-
-	if step.Started == 0 && step.Stopped != 0 {
-		step.Started = started
-	}
```

Comment:
Could you explain why this isn't needed?

Reply:
Because of the check `if step.Stopped == 0`

without the check there are edgecases where could be the case a stoped steped can be markt as running again, witch is wrong.

I do remember we have "running" steps indefinetly in cancled pipelines. This could be the fix, i just did not test that specific.

Anyway the func hat a good testcoverage ... so just look at the tests

Reply:
I'll split it out into a new pull